cnf-certification-test/accesscontrol/suite.go

```diff
@@ -17,12 +17,294 @@
 
 import (
+	"strings"
+
 	"github.com/onsi/ginkgo/v2"
+	"github.com/onsi/gomega"
 	"github.com/sirupsen/logrus"
 	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/common"
+	"github.com/test-network-function/cnf-certification-test/cnf-certification-test/identifiers"
+	"github.com/test-network-function/cnf-certification-test/pkg/provider"
+	"github.com/test-network-function/cnf-certification-test/pkg/tnf"
+)
+
+var (
+	nonCompliantCapabilites = []string{"NET_ADMIN", "SYS_ADMIN", "NET_RAW", "IPC_LOCK"}
 )
 
 var _ = ginkgo.Describe(common.AccessControlTestKey, func() {
 
-	logrus.Debugf("%s not moved yet to new framework", common.AccessControlTestKey)
+	logrus.Debugf("Entering %s suite", common.AccessControlTestKey)
+	var env provider.TestEnvironment
+	ginkgo.BeforeEach(func() {
+		provider.BuildTestEnvironment()
+		env = provider.GetTestEnvironment()
+	})
+	// Security Context: non-compliant capabilities
+	testID := identifiers.XformToGinkgoItIdentifier(identifiers.TestSecConCapabilitiesIdentifier)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestSecConCapabilities(&env)
+	})
+	// container security context: non-root user
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestSecConNonRootUserIdentifier)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestSecConRootUser(&env)
+	})
+	// container security context: privileged escalation
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestSecConPrivilegeEscalation)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestSecConPrivilegeEscalation(&env)
+	})
+	// container security context: host port
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestContainerHostPort)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestContainerHostPort(&env)
+	})
+	// container security context: host network
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodHostNetwork)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestPodHostNetwork(&env)
+	})
+	// pod host path
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodHostPath)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestPodHostPath(&env)
+	})
+	// pod host ipc
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodHostIPC)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestPodHostIPC(&env)
+	})
+	// pod host pid
+	testID = identifiers.XformToGinkgoItIdentifier(identifiers.TestPodHostPID)
+	ginkgo.It(testID, ginkgo.Label(testID), func() {
+		TestPodHostPID(&env)
+	})
 })
+
+// TestSecConCapabilities verrifies that non compliant capabilities are not present
+func TestSecConCapabilities(env *provider.TestEnvironment) {
+	var badContainers []string
+	var errContainers []string
+	if len(env.Containers) == 0 {
+		ginkgo.Skip("No containers to perform test, skipping")
+	}
+	for _, cut := range env.Containers {
+		if cut == nil {
+			errContainers = append(errContainers, cut.Podname+"."+cut.Data.Name)
```

Comment:
nice catch, this would crash if the cut is nil

Reply:
Also, this seems very repetitive, we want our cut not to be nil in all the tests..
should we have one function that tests that ? otherwise, this check would be repeated in all the tests. WDYK ?

Reply:
another thing, is by construction cut would never be nil. We do append only valid cut to environment test.

Reply:
Thanks, I will remove the test

Reply:
fixed